x-pack/plugins/maps/public/layers/layer_wizard_registry.ts

```diff
@@ -6,13 +6,17 @@
 /* eslint-disable @typescript-eslint/consistent-type-definitions */
 
 import { ReactElement } from 'react';
-import { ISource } from './sources/source';
-
-export type PreviewSourceHandler = (source: ISource | null) => void;
+import { LayerDescriptor } from '../../common/descriptor_types';
 
 export type RenderWizardArguments = {
-  onPreviewSource: PreviewSourceHandler;
-  inspectorAdapters: object;
```

Comment:
ok just transferring this API to here. but this looks complex and probably takes us further from https://github.com/elastic/kibana/issues/60918, especially when thinking of improving plugability.

(but can investigate how we can simplify later).